backend/manager/modules/utils/src/main/java/org/ovirt/engine/core/utils/ovf/OvfOvirtReader.java

```diff
@@ -146,6 +146,16 @@ private XmlNode getNode(XmlNodeList nodeList, String attributeName, String attri
         return null;
     }
 
+    protected String getInstanceIdValue(XmlNode node) {
+        XmlNode candidateNode = selectSingleNode(node, "rasd:InstanceID", _xmlNS);
+        //Also there is a defect that has key InstanceID as InstanceId in old oVirt releases
```

Comment:
that's not accurate - we have two kind of OVFs:
1. OVFs within OVAs - that's the "newer format" that we tried to align to the common standard as much as possible. in bz 1850103, you pointed out that we need to write InstanceID and that's what we do since that bug was fixed

2. "internal OVFs" - OVFs that are stored within data domains (OVF STOREs), snapshots, export domains. that format is not perfectly aligned with the standard. from the description of this PR it seems you refer to this format and this class indeed handles that. however, ovirt never wrote InstanceID there in order to preserve backward compatibility. we can change it but we'll have to support both until versions that use InstanceId are not suppported anymore - and in my opinion, it doesn't worth the trouble, those OVFs are only used internally, no other platform should use them..
I have a vague memory of an issue I looked at that made me think Oracle's product writes InstanceID in those OVFs, could it be that Oracle's product is not aligned with ovirt on this and that's why you find InstanceID in those OVFs?

Reply:
Thank you Arik Hadas for the review and comments!
I have made the changes based on the feedback and updated the PR.